athena/active.py

```diff
@@ -69,7 +69,9 @@ def fit(self,
                                                outputs=outputs,
                                                weights=weights)[0]
 
-        if generator is None:
+        if isinstance(gradients, types.GeneratorType):
+            self.singvals, self.evects = self._frequent_directions(gradients)
```

Comment:
here you should return the eigenvalues, otherwise you cannot use all the other methods already implemented.